src/main/scala/spray/json/ProductFormats.scala

```diff
@@ -152,3 +152,14 @@ trait NullOptions extends ProductFormats {
     (fieldName, writer.write(value)) :: rest
   }
 }
+
+/**
+ * This trait changes the behavior for reading JSON values.
+ * If you mix in this trait into your custom JsonProtocol, JSON deserialization
+ * will throw an error if the input contains keys that are not a field of the
+ * target case class.
+ */
+trait ExtraKeysOptions extends ProductFormats {
```

Comment:
Do we actually need that trait? Seems a bit weird API to just change the flag from `true` to `false`. Maybe just document `allowExtraKeys` directly.